docs/docs/spec.md

```diff
@@ -135,9 +135,9 @@ A nullness operator is one of 4 values:
 >         etc., plus `null`.
 >     -   The type-variable usage `T UNION_NULL` includes all members of `T`,
 >         plus `null` if it wasn't already included.
-> -   `NO_CHANGE`: This is the operator produced by *not* putting `@Nullable` on
->     a type usage (aside from the exception discussed under `UNSPECIFIED`
->     below).
+> -   `NO_CHANGE`: This is the operator produced by putting neither `@Nullable`
```

Comment:
It might be nicer to define the operators' meanings without referring to how to determine which operator applies based on annotations, which is in a later section.

* `UNION_NULL` means that `null` is a valid value.
* `NO_CHANGE` means the same as `MINUS_NULL` except for type variables, where it means that `null` is a valid value only if it is for the substituted type argument.
* `UNSPECIFIED` means that we don't know whether `null` is a valid value.
* `MINUS_NULL` means that `null` is not a valid value.

Reply:
Here, I'm kind of throwing a bunch of explanations at the wall to see what will stick for any given person. It's true that this explanation is a sort of forward reference, but I think it would be overly coy to _not_ yell "`UNION_NULL` is `@Nullable`!" (The mention is `@Nullable` is probably especially welcome after the lecture I give above about how I'm not going to talk about a type "is nullable" :)) I do think that the sort of explanation that you give is also valuable, and I have an ancient attempt at that in the sub-bullets.

Do you think that reshuffling and tweaking what we currently have could improve matters? Is there just _too much_ here?

Reply:
I think `@Nullable` and `@NonNull` represent things that people have seen before, so using that to signify what those operators mean makes sense. I think saying "completely unannotated code" is helpful for `UNSPECIFIED`.

But I don't think that this statement for `NO_CHANGE` will be helpful to someone who hasn't read the rest of the spec.

Maybe part of the problem is that we're using these operators in two ways: as a component of an augmented type (base type + operator) and as an input to the substitution rule (where we apply a nullness operator to an augmented type to produce a new augmented type). `NO_CHANGE` is not obvious in the first case, but makes a lot of sense in the second. That's part of why I'd like to change the rules so that `NO_CHANGE` only applies to type variable usages (and, I guess, intersection types). But we don't have to do that for 1.0.

Reply:
Thanks, that's very helpful! Now you get to be the judge of whether I put that helpful comment to good use :)